### DISCUSSION TITLE
Using OTLP in a forking model application

### DISCUSSION THREAD

**Discussion by NickPoole** [open-telemetry/opentelemetry-cpp#3047] [Q&A]
I'm trying to incorporate OTLP into an existing server application that forks the process to handle each client request.
So far I've found that gRPC is a no-go without rebuilding with fork support, but even then I'm not sure how to enable this through opentelemetry-cpp. HTTP seems to work fine for a while, but under enough load the number of ephemeral sockets is exhausted because of time wait (I haven't worked out if the library is letting curl maintain 1 connection per-process yet or if a new connection is made per log entry, but I'm suspecting the former).

Are there any known strategies for using the library in forking applications?

**Comment by malkia** [comment]
I wonder if using file pipes with grpc exporter might work, piping to the main grandparent... but no idea if it'll work. Also now that you'll have split brain, you would probably have to add additional attribute (process_vpid? of sorts) to each fork of your apps, so that cumulative metrics work.
